integration-test/src/test/java/org/apache/iotdb/db/it/query/IoTDBLoadEncryptedTsFileIT.java

```diff
@@ -0,0 +1,200 @@
+        Statement statement = connection.createStatement()) {
+      statement.execute("CREATE DATABASE root.tesgsg");
+      statement.execute("CREATE TIMESERIES root.testsg.d1.s1 WITH DATATYPE=INT32, ENCODING=PLAIN");
+      File tsfile = generateSameWayEncryptedFile();
+      statement.execute(String.format("load \"%s\"", tsfile.getParentFile().getAbsolutePath()));
+      ResultSet resultSet = statement.executeQuery("select s1 from root.testsg.d1");
+      ResultSetMetaData resultSetMetaData = resultSet.getMetaData();
+      List<Integer> actualIndexToExpectedIndexList =
+          checkHeader(
+              resultSetMetaData,
+              "Time,root.testsg.d1.s1,",
+              new int[] {
+                Types.TIMESTAMP, Types.INTEGER,
+              });
+
+      int cnt = 0;
+      while (resultSet.next()) {
+        String[] expectedStrings = retArray[cnt].split(",");
+        StringBuilder expectedBuilder = new StringBuilder();
+        StringBuilder actualBuilder = new StringBuilder();
+        for (int i = 1; i <= resultSetMetaData.getColumnCount(); i++) {
+          actualBuilder.append(resultSet.getString(i)).append(",");
+          expectedBuilder
+              .append(expectedStrings[actualIndexToExpectedIndexList.get(i - 1)])
+              .append(",");
+        }
+        Assert.assertEquals(expectedBuilder.toString(), actualBuilder.toString());
+        cnt++;
+      }
+      Assert.assertEquals(3, cnt);
+    } catch (Exception e) {
+      Assert.fail(e.getMessage());
+    }
+  }
+
+  @Test
+  public void loadAnotherWayEncryptedTsFileTest() {
+    String unrecognizedType = "org.apache.tsfile.encrypt.SM4128";
+    try (Connection connection = EnvFactory.getEnv().getConnection();
+        Statement statement = connection.createStatement()) {
+      statement.execute("CREATE DATABASE root.tesgsg1");
+      statement.execute("CREATE TIMESERIES root.testsg1.d1.s1 WITH DATATYPE=INT32, ENCODING=PLAIN");
+      File tsfile = generateAnotherWayEncryptedFile(unrecognizedType);
+      statement.execute(String.format("load \"%s\"", tsfile.getParentFile().getAbsolutePath()));
+      ResultSet resultSet = statement.executeQuery("select s1 from root.testsg1.d1");
+      Assert.fail();
+    } catch (Exception e) {
+      System.out.println(e.getMessage());
+      Assert.assertTrue(e.getMessage().contains(unrecognizedType));
+    }
+  }
+
+  private File generateSameWayEncryptedFile() throws IOException {
+    Path tempDir = Files.createTempDirectory("");
+    tempDir.toFile().deleteOnExit();
+    String tsfileName =
+        TsFileNameGenerator.generateNewTsFileName(System.currentTimeMillis(), 1, 0, 0);
+    File tsfile = new File(tempDir + File.separator + tsfileName);
+    Files.createFile(tsfile.toPath());
+    TSFileConfig config = TSFileDescriptor.getInstance().getConfig();
+    config.setEncryptFlag("true");
+    config.setEncryptType("org.apache.tsfile.encrypt.AES128");
```

Comment:
The configs should be rolled back after writing the file. (Maybe in the finally clause)